### PR TITLE
Added error checks.

### PR DESCRIPTION
**Pull request by peturingi**
Added error checks.
Removed a confusing variable 'current' from addAll(Collection<Integer> list).

In 'int get(int index)', it might be more robust to invert fromIndex and toIndex.

### CODE REVIEW

File: runtime/Java/src/org/antlr/v4/runtime/misc/IntegerList.java [hunk 1]
```diff
@@ -57,6 +58,8 @@ public IntegerList(int capacity) {
 		if (capacity < 0) {
 			throw new IllegalArgumentException();
 		}
+        if (capacity > MAX_ARRAY_SIZE)
+          throw new IllegalArgumentException();
```
Comment (sharwell) [inline runtime/Java/src/org/antlr/v4/runtime/misc/IntegerList.java]: :bulb: Considering `ArrayList` omits this check, I think it's safe for us to as well.

File: runtime/Java/src/org/antlr/v4/runtime/misc/IntegerList.java [hunk 2]
```diff
@@ -100,13 +103,9 @@ public final void addAll(IntegerList list) {
...
 	public final void addAll(Collection<Integer> list) {
 		ensureCapacity(_size + list.size());
-		int current = 0;
 		for (int x : list) {
-			_data[_size + current] = x;
-			current++;
-		}
-
-		_size += list.size();
+          _data[_size++] = x;
```
Comment (sharwell) [inline runtime/Java/src/org/antlr/v4/runtime/misc/IntegerList.java]: :exclamation: Even though it's not documented (and the class is not safe for concurrent operations on multiple threads), the original implementation is saf*er* for single-writer, multiple-reader scenarios. The new implementation contains a sequence point where `list.get(list.size() - 1)` returns an uninitialized value, even if memory fences are employed.

File: runtime/Java/src/org/antlr/v4/runtime/misc/IntegerList.java [hunk 3]
```diff
@@ -268,6 +267,8 @@ public final int binarySearch(int fromIndex, int toIndex, int key) {
 		if (fromIndex < 0 || toIndex < 0 || fromIndex > _size || toIndex > _size) {
 			throw new IndexOutOfBoundsException();
 		}
+        if (fromIndex > toIndex)
```
Comment (sharwell) [inline runtime/Java/src/org/antlr/v4/runtime/misc/IntegerList.java]: I think this condition makes sense. Can you update the code to match the surrounding formatting (hard tabs, and use braces)?

Comment (peturingi) [inline runtime/Java/src/org/antlr/v4/runtime/misc/IntegerList.java]: Yes.

**Comment by parrt** [comment]
Hi. we'll need you to add yourself to https://github.com/antlr/antlr4/blob/master/contributors.txt

**Comment by peturingi** [comment]
Yes, done.

**Pull request merged**